resources/schema.py

```diff
@@ -452,6 +461,55 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         return DeleteHarborMutation()
 
 
+class PierInput(AbstractAreaSectionInput):
+    harbor_id = graphene.ID()
+    suitable_boat_types = graphene.List(graphene.String)
+    mooring = graphene.Boolean()
+    waste_collection = graphene.Boolean()
+    lighting = graphene.Boolean()
+
+
+class CreatePierMutation(graphene.ClientIDMutation):
+    class Input(PierInput):
+        harbor_id = graphene.ID(required=True)
+
+    pier = graphene.Field(PierNode)
+
+    @classmethod
+    @login_required
+    @superuser_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO: Should check if the user has permissions to
+        # delete the specific resource
+        suitable_boat_types = kwargs.pop("suitable_boat_types", [])
+
+        harbor_id = kwargs.pop("harbor_id", None)
+        if harbor_id:
+            harbor_global_id = from_global_id(harbor_id)[1]
+            try:
+                kwargs["harbor"] = Harbor.objects.get(pk=harbor_global_id)
+            except Harbor.DoesNotExist as e:
+                raise VenepaikkaGraphQLError(e)
+
+        boat_types = set([])
+        for boat_type_id in suitable_boat_types:
+            try:
+                boat_type = BoatType.objects.get(pk=boat_type_id)
```

Comment:
would this work:
```python
suitable_boat_types = kwargs.pop("suitable_boat_types", [])

boat_types = []
try:
    boat_types = BoatType.objects.get(pk__in=suitable_boat_types)
except BoatType.DoesNotExist as e:
    raise VenepaikkaGraphQLError(e)

...
```

Reply:
Yeah, would work if there was a way to figure out if there was an invalid one.

Reply:
yeah, ok, let's keep it like this

Reply:
Another option would be to compare something like this:
```python
set(suitable_boat_types).difference(
    set(Model.objects.filter(id__in=suitable_boat_types).values_list('id'),
    flat=True)
)
```
And if there's a difference, we can raise the error saying that values X, Y, &Z  are missing.

Reply:
oh, pretty neat idea! probably a bit of an overengineering for this case, and this is still a hit to the db, but slightly less readable that your current solution. let's keep things as they are 🙂